### ISSUE TITLE
Bitcoin client package

### ISSUE THREAD

**Issue by haxwithaxe** [Byzantium/Byzantium#175]
in #173 suggestion to include bitcoin for use post SHTF once internet access is restored.
i suggest it should wait for v2.0 development if included.
comments requested ...

**Comment by virtadpt** [comment]
There are a few things that need to be taken into account.  The first is whether or not the Bitcoin client will store the blockchain locally.  The blockchain (which totaled out at 5.2 GiB as of 2 December 2012; I don't know how big it is now) takes a significant amount of time to download.  The size of the blockchain at this point in time is likely too large to fit on a system running a live distro unless it has around eight gigs of RAM.  There is also the fact that there may not be enough bandwidth to download the blockchain through a mesh gateway at this point in time.  There was two years ago, but that might not be the case now.

There are some implementations of Bitcoin that do not store the blockchain locally, but instead reference one of the public blockchain servers.  Multibit, for example, does this.  However, Multibit is written in Java, which means that we'd have to package Java along with it.

There is, however, a simpler solution: A web-based Bitcoin wallet, for example https://easywallet.org/

The idea is that the public keys for all Bitcoin accounts are in the blockchain, and it's the private keys that allow anyone with a copy of the blockchain to recover their Bitcoins.  Easy Wallet gives you the private key in the form of a URL that you're supposed to bookmark or copy into a file, and the public key (the address) appears in the form of a string of text and a QR code that can be scanned by a mobile device.  They charge a nominal fee for transactions.

Another web-based Bitcoin wallet is https://blockchain.info/wallet/ (which I am a user of).  You have to set up an account (which is trivial) but they don't charge you anything, and in fact their big selling point for customers is that all you need is a smartphone to carry out transactions (they did this because Apple regularly hunts down and culls Bitcoin implementations from the Appstore).

tl;dr: It would make more sense to bookmark web-based Bitcoin wallets than it would to include software implementations of Bitcoin.  They're even usable over Tor (if and when we get around to the gateway-over-Tor part).

**Comment by haxwithaxe** [comment]
sneakernet blockchain bootstrap perhaps? not exactly automatic but it would work better than shifting that much data to every byzantium node.

**Comment by ghost** [comment]
A web-based Bitcoin wallet could be a easy solution, just add a bookmark to browser.
Do this for each version of Byzantium.

**Comment by virtadpt** [comment]
That's sort of what I was thinking.  It's much less CPU power that the node has to employ and much less bandwidth that will be used.

**Comment by virtadpt** [comment]
Won't fix - way outside of our use case.  Not to say that it's not useful - Cyprus comes immediately to mind - but folks that are going to use Bitcoin probably already know of web wallets or already have Bitcoin nodes, and there isn't much we can offer them besides potential connectivity.
